elide-graphql/src/main/java/com/yahoo/elide/graphql/ModelBuilder.java

```diff
@@ -394,14 +395,23 @@ private GraphQLInputType buildInputObjectStub(Class<?> clazz) {
 
             /* If the attribute is an object, we need to change its name so it doesn't conflict with query objects */
             if (attributeType instanceof GraphQLInputObjectType) {
-                MutableGraphQLInputObjectType wrappedType =
-                    new MutableGraphQLInputObjectType(
-                        attributeType.getName() + ARGUMENT_INPUT,
-                            ((GraphQLInputObjectType) attributeType).getDescription(),
-                            ((GraphQLInputObjectType) attributeType).getFields()
-                        );
-                attributeType = wrappedType;
-
+                String objectName = attributeType.getName() + ARGUMENT_INPUT;
+                if (!convertedInputs.containsKey(objectName)) {
+                    MutableGraphQLInputObjectType wrappedType =
+                            new MutableGraphQLInputObjectType(
+                                    entityName + "__" + attributeType.getName() + ARGUMENT_INPUT,
```

Comment:
This seems wrong to me.  If the type is _Address_.  I would expect the name to not be "author__address" or "publisher__address".  The name should be linked to the type - not the attribute name.

Reply:
I agree with this. I'll have to revisit. The types are mostly important for schema viewing and query variables. The case for query variables, however, is far more important imo and I believe we should optimize for those.

Before we merge this PR I should have several `GraphQLIT` tests that have query mutation variables so we can have a nice example of how it works.